src/pages/VeSDL/index.tsx

```diff
@@ -10,48 +10,228 @@ import {
+    }))
+    setUnlockDate(null)
+  }
+
+  const currentTimestamp = getUnixTime(new Date())
+  const unlockDateOrLockEnd = unlockDate || lockEnd
+  const expireTimestamp =
+    unlockDateOrLockEnd && !isNaN(unlockDateOrLockEnd.valueOf())
+      ? getUnixTime(unlockDateOrLockEnd)
+      : undefined
+  const totalAmount =
+    !sdlTokenValue.isZero() || unlockDate
+      ? sdlTokenValue.add(lockedSDLVal)
+      : Zero
+
+  // Calculate estimated veSdl amount
+  const estLockAmt = expireTimestamp
+    ? totalAmount
+        .mul(
+          BigNumber.from(expireTimestamp).sub(BigNumber.from(currentTimestamp)),
+        )
+        .div(BigNumber.from(MAXTIME))
+    : Zero
+
+  // Calculate penalty Ratio and penalty amount
+  const leftTimeForUnlock = lockEnd && getUnixTime(lockEnd) - currentTimestamp
+  const penaltyAmount = leftTimeForUnlock
+    ? minBigNumber(
+        lockedSDLVal.mul(BigNumber.from(3)).div(BigNumber.from(4)),
+        lockedSDLVal
+          .mul(BigNumber.from(leftTimeForUnlock))
+          .div(BigNumber.from(MAXTIME)),
+      )
+    : Zero
+
+  const handleLock = async () => {
+    const unlockTimeStamp = unlockDate ? getUnixTime(unlockDate) : null
+    const hasLockedSDL = !lockedSDLVal.isZero()
+    const shouldCreateLock =
+      !hasLockedSDL && sdlTokenValue.gt(Zero) && unlockTimeStamp
+    const shouldIncreaseAmoutAndLockEnd =
+      hasLockedSDL && sdlTokenValue.gt(Zero) && unlockTimeStamp
+    const shouldIncreaseAmount =
+      hasLockedSDL && sdlTokenValue.gt(Zero) && !unlockTimeStamp
+    const shouldIncreaseLockEnd =
+      hasLockedSDL && sdlTokenValue.isZero() && unlockTimeStamp
+
+    if (!account || !chainId || !votingEscrowContract?.address) return
+    try {
+      if (sdlTokenValue.gt(Zero)) {
+        await checkAndApproveTokenForTrade(
+          sdlContract,
+          votingEscrowContract.address,
+          account,
+          sdlTokenValue,
+          infiniteApproval,
+        )
+      }
+
+      if (shouldCreateLock && unlockTimeStamp) {
+        const txn = await votingEscrowContract?.create_lock(
+          sdlTokenValue,
+          unlockTimeStamp,
+        )
+        void enqueuePromiseToast(chainId, txn.wait(), "createLock")
+      } else if (shouldIncreaseAmoutAndLockEnd) {
+        const txnIncreaseAmount = await votingEscrowContract.increase_amount(
+          sdlTokenValue,
+        )
+        await txnIncreaseAmount.wait()
+        const txnIncreaseUnlockTime =
+          await votingEscrowContract.increase_unlock_time(
+            BigNumber.from(unlockTimeStamp),
+          )
+        await txnIncreaseUnlockTime.wait()
+        enqueueToast("success", "Increased amount and unlock time")
+      } else if (shouldIncreaseAmount) {
+        // Deposit additional SDL into and existing lock
+        const txn = await votingEscrowContract.increase_amount(sdlTokenValue)
+        await txn.wait()
+        void enqueuePromiseToast(chainId, txn.wait(), "increaseLockAmt")
```

Comment:
Why is `enqueueToast` used in some places, and `enqueuePromiseToast` used in others? 
Why do some blocks await `txn.wait()`, while others pass the promise directly to `enqueuePromiseToast` without await-ing?

Reply:
Enqueue promise toast captures pending and success/fail states. enqueueToast is for other scenarios when there isn't a necessarily a promise based action happening. Having both in one try/catch might be unnecessary i forget.